docs/serviceability.md

```diff
@@ -0,0 +1,76 @@
+```
+
+To understand the AppStudio controller logging convention, refer to the Appstudio [ADR](https://github.com/redhat-appstudio/book/blob/main/ADR/0006-log-conventions.md)
+
+## Debugging
+
+- Insert break points at the controller functions to debug unit tests
+- When debugging unit and integration tests, remember that the mock clients used are hosting dummy data and mocking the api call and returns
+- For debugging controller logs either from a local cluster or a managed cluster, gather the log and search for:
+  - the namespace the user belongs to. For example, if the user is on the project/namespace, search for `user-tenant`
+  - the resource name that the controller is reconciling. For example, if the `SnapshotEnvironmentBinding` resource is being reconciled, then search for `"name":"devfile-sample-go-basic-development-binding-hr9nm"` belonging to `"controllerKind":"SnapshotEnvironmentBinding"`
+  - the log message. For example, `"msg":"Finished reconcile loop for user-tenant/devfile-sample-go-basic-development-binding-hr9nm"`. You may search for the string `Finished reconcile loop for` in the application-service repository to track down the code logic that is emitting the log. Remember to look out for resource name concatenation and/or error wrapping that may not turn up in your code search. It is advised to exclude such strings from the code search for debugging purposes
+
+## Common Problems
+- When deploying HAS locally or on a local cluster, a Github Personal Access Token is required as the application-service controller requires the token for pushing the resources to the GitOps repository. Please refer to the [instructions](../docs/deploy.md#creating-a-github-secret-for-application-service) in the deploy section for more information
+- When creating a `Component` from the `ComponentDetectionQuery`, remember to replace the generic application name `insert-application-name`, if the information is being used from a `ComponentDetectionQuery` status
+
+## FAQs
+Q. Where can I view the application-service api types?
+
+A. The application-service api types and their corresponding webhooks are defined in the [redhat-appstudio/application-api](https://github.com/redhat-appstudio/application-api) repository.
+
+Q. Where are the application-service controller logic located?
+
+A. Most of the application-service business logic like the reconcile functions are located in the [controllers](https://github.com/redhat-appstudio/application-service/tree/main/controllers) pkg.
+
+Q. What is the application-service release process?
+
+A. The release process for application-service is yet to be determined and defined.
```

Comment:
Since HAS is just part of RHTAP / AppStudio, I would just saying that commits made to this repo automatically get added to infra deployments

Reply:
oops